website/docs/workloads/hadoop/hadoop.md

```diff
@@ -0,0 +1,136 @@
+# TeraSort: Sorting at Scale
+
+## Overview
+
+TeraSort is a benchmark used to evaluate the efficiency of distributed computing frameworks in sorting large datasets. The benchmark was introduced by the Apache Hadoop project, which is a popular open-source framework for distributed storage and processing of big data. TeraSort is designed to test the scalability and performance of distributed systems when sorting massive amounts of data.
+
+[Official Documentation](https://hadoop.apache.org/docs/r3.3.5/hadoop-project-dist/hadoop-common/SingleCluster.html)
+
+## Why TeraSort?
+
+Sorting is a fundamental operation in data processing, and it becomes especially challenging when dealing with enormous datasets that cannot fit into the memory of a single machine. TeraSort addresses the need for efficient sorting in distributed environments and helps assess the performance of big data processing frameworks.
+
+### Key Objectives of TeraSort
+
+1. **Scalability:** TeraSort aims to evaluate how well a distributed computing framework scales when sorting data across a large number of nodes or machines.
+
+2. **Efficiency:** The benchmark assesses the efficiency of the sorting algorithm and the overall data processing framework in terms of time and resource utilization.
+
+3. **Fault Tolerance:** TeraSort provides insights into how well a system can handle failures or node outages during the sorting process, crucial for robustness in large-scale distributed systems.
+
+4. **Optimization:** It encourages developers to optimize sorting algorithms and data processing techniques to achieve better performance on distributed platforms.
+
+## How TeraSort Works
+
+TeraSort uses a specific dataset generated with a total size of one terabyte, hence the name. The dataset consists of key-value pairs, where the key is a 10-byte long random string, and the value is a 90-byte payload. The sorting task involves ordering the key-value pairs based on the keys.
+
+The sorting process is divided into map and reduce phases:
+
+1. **Map Phase:** Each node in the distributed system processes a portion of the input dataset and produces a set of key-value pairs with the keys as the sorting criterion.
+
+2. **Shuffle and Sort Phase:** The framework shuffles the intermediate key-value pairs, ensuring that all values associated with the same key are sent to the same reducer.
+
+3. **Reduce Phase:** Each reducer processes a subset of the sorted key-value pairs, producing the final globally sorted output.
+
+## Support for Java versions
+
+The Hadoop 3.3.x versions support Java 8 and Java 11.
+
+Adding the reference of the documentation here: [Supported Java Versions for Hadoop](https://cwiki.apache.org/confluence/display/HADOOP/Hadoop+Java+Versions)
+
+## What is Being Measured?
+
```

Comment:
Please mention the most important metrics

Reply:
I have listed all of the metrics below. Based on the usecase, I will make the changes if required.